external/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# -*- mode: cmake; -*-
+# - osquery external project builds
+#
+# Symlink directories for extensions or modules here.
+# If a directory includes: "extension_NAME" it is built as an extension.
+# If a directory includes: "module_NAME" it is built as a module.
+
+add_custom_target(externals)
+add_definitions(-DOSQUERY_EXTERNAL)
+
+macro(SUBDIRLIST OUTPUT PWD)
```

Comment:
Macros will pollute the current scope with variables; when possible, we should use functions instead

Reply:
For the same reason, we always have a main function in the file (in this case it would be called externalsMain